cantools/database/can/environment_variable.py

```diff
@@ -0,0 +1,161 @@
+# A CAN environment variable.
+
+class EnvironmentVariable(object):
+    """A CAN environment variable.
+
+    """
+
+    def __init__(self,
+                 name,
+                 env_type,
+                 minimum,
+                 maximum,
+                 unit,
+                 initial_value,
+                 env_id,
+                 access_type,
+                 access_node,
+                 comment):
+        self._name = name
+        self._type = env_type
+        self._minimum = minimum
+        self._maximum = maximum
+        self._unit = unit
+        self._initial_value = initial_value
+        self._id = env_id
+        self._access_type = access_type
+        self._access_node = access_node
+        self._comment = comment
+
+    @property
+    def name(self):
+        """The environment variable name as a string.
+
+        """
+
+        return self._name
+
+    @name.setter
+    def name(self, value):
+        self._name = value
+
+    @property
+    def type(self):
+        """The environment variable type as a string.
+
+        """
+
+        return self._type
+
+    @type.setter
+    def type(self, value):
+        self._type = value
+
+    @property
+    def minimum(self):
+        """The environment variable minimum as a string.
+
+        """
+
+        return self._minimum
+
+    @minimum.setter
+    def minimum(self, value):
+        self._minimum = value
+
+    @property
+    def maximum(self):
+        """The environment variable maximum as a string.
+
+        """
+
+        return self._maximum
+
+    @maximum.setter
+    def maximum(self, value):
+        self._maximum = value
+
+    @property
+    def unit(self):
+        """The environment variable unit as a string.
+
+        """
+
+        return self._unit
+
+    @unit.setter
+    def unit(self, value):
+        self._unit = value
+
+    @property
+    def initial_value(self):
+        """The environment variable initial_value as a string.
```

Comment:
This is not a string according to the test case. Same goes for minimum, maximum, and other members.

Reply:
done

Reply:
I meant the docstring was wrong. The code itself was ok. There is no need to call ``num`` or ``int`` anywhere in this file. At least they are not called in the similar classes ``AttributeDefinition`` or ``Attribute``.

Reply:
Yes, the docstring is already changed.
About the `int`and `num` calls, is it not the class's responsibility to ensure the integrity of the values it houses?

Reply:
Let's make this class consistent with other classes for now, and maybe change all classes at once later.

Reply:
Right, then, so be it :+1: